### ISSUE TITLE
New Package Request: [open-vm-tools-containerinfo]

### ISSUE THREAD

**Issue by Winnie81** [flatcar/Flatcar#781]
As https://github.com/vmware/open-vm-tools/blob/stable-12.0.0/ReleaseNotes.md#continfo described, VMware Tools 12.0.0 supports for gathering and publishing list of containers running inside Linux guests.  Checked latest alpha build 3255.0.0, open-vm-tools-containerinfo is missing.

Could you please help add it? Thanks a lot.

**Comment by tormath1** [comment]
Hi @Winnie81 thanks for the follow-up on this package. Quickly looking at the configuration, this plugin is enabled in `auto` mode - if the required dependencies are found, the plugin is built: https://github.com/vmware/open-vm-tools/blob/b7ab96e6e38f6559cb882278a53d7a7ea5c0592c/open-vm-tools/configure.ac#L660-L661.

From a quick look, missing build/runtime dependency for Flatcar is `grpc` (`net-libs/grpc`) - we could try to configure and install `grpc` in the OEM partition and to configure `open-vm-tools` to consume it (like with `dev-libs/libdnet`).

What do you think @dongsupark ?

**Comment by dongsupark** [comment]
@tormath1 Yes, that sounds correct.
Though I am not sure how much free space we have in the OEM partition.

**Comment by tormath1** [comment]
To give an update - I can successfully install grpc in the OEM partition (with a few tricks to build `grpc_cpp_plugin`). There is now this thing to solve: https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/configure.ac#L726-L755

Not sure how to manage this dependency.

**Comment by krnowak** [comment]
Probably by a patch in FILESDIR that adds those files somewhere in the workdir, and also patches the configure script to search those proto files in our directory, not `/usr/share/gocode/whatever`.

**Comment by tormath1** [comment]
Yeah - I wanted to avoid that because we're missing the versioning of these file. I don't see other options at the moment.

**Comment by jepio** [comment]
These are development files, I would install them to the expected path since they are not needed at runtime, so don't need to be in the OEM partition.

**Comment by tormath1** [comment]
@jepio I'm not sure I can install them in the expected path - since they need to be present at the `open-vm-tools` compilation time I won't be able to install them properly from `${FILESDIR}` to `/usr/share/gocode/whatever`

**Comment by krnowak** [comment]
> @jepio I'm not sure I can install them in the expected path - since they need to be present at the `open-vm-tools` compilation time I won't be able to install them properly from `${FILESDIR}` to `/usr/share/gocode/whatever`

You could do that by adding separate package (`dev-util/proto-files-for-open-vm-tools`), but let's not do that, ok? :)

**Comment by t-lo** [comment]
Minor update: Implementation currently suffers from
1. the fact that we need to pull in dependency libraries (`grpc` and friends) and 
2. that we are required to make these libraries work in `/usr/share/oem`, so dependencies are only included with the VMWare OEM image.

We believe that adding sysext support (see #548 ) will resolve this and other (e.g. #60) issues. Leaving this issue open for tracking.
